yb_migrate/cmd/importDataFileCommand.go

```diff
@@ -0,0 +1,264 @@
+		TableRowCount: tableFileSize,
+		Delimiter:     delimiter,
+		HasHeader:     hasHeader,
+		ExportDir:     exportDir,
+	}
+	dfd.Save()
+}
+
+func getFileSizeInfo() map[string]int64 {
+	tableFileSize := make(map[string]int64)
+	for table, filePath := range tableNameVsFilePath {
+		fileInfo, err := os.Stat(filePath)
+		if err != nil {
+			utils.ErrExit("calculating file size of %q in bytes: %v", filePath, err)
+		}
+		tableFileSize[table] = int64(fileInfo.Size())
+
+		log.Infof("File size of %q for table %q: %d", filePath, table, tableFileSize[table])
+	}
+
+	return tableFileSize
+}
+
+func createDataFileSymLinks() {
+	log.Infof("creating symlinks to original data files")
+	for table, filePath := range tableNameVsFilePath {
+		symLinkPath := filepath.Join(exportDir, "data", table+"_data.sql")
+
+		filePath, err := filepath.Abs(filePath)
+		if err != nil {
+			utils.ErrExit("absolute original filepath for table %q: %v", table, err)
+		}
+		log.Infof("absolute filepath for %q: %q", table, filePath)
+		log.Infof("symlink path for file %q is %q", filePath, symLinkPath)
+		if utils.FileOrFolderExists(symLinkPath) {
+			resolvedFilePath, err := os.Readlink(symLinkPath)
+			if err != nil {
+				utils.ErrExit("resolving symlink %q: %v", symLinkPath, err)
+			}
+			if resolvedFilePath == filePath {
+				log.Infof("using symlink %q already present", symLinkPath)
+				continue
+			} else {
+				log.Infof("removing symlink: %q to create fresh link", symLinkPath)
+				err = os.Remove(symLinkPath)
+				if err != nil {
+					utils.ErrExit("removing symlink %q: %v", symLinkPath, err)
+				}
+			}
+		}
+
+		err = os.Symlink(filePath, symLinkPath)
+		if err != nil {
+			utils.ErrExit("error creating symlink to data file %q: %v", filePath, err)
+		}
+	}
+}
+
+func prepareCopyCommands() {
+	log.Infof("preparing copy commands for the tables to import")
+	for table := range tableNameVsFilePath {
+		copyCommand := fmt.Sprintf(`COPY %s FROM STDIN DELIMITER '%c'`, table, []rune(delimiter)[0])
+		if hasHeader {
+			copyCommand += " CSV HEADER"
+		}
+		copyTableFromCommands[table] = copyCommand
+	}
+
+	log.Infof("copyTableFromCommands map: %+v", copyTableFromCommands)
+}
+
+func setImportTableListFlag() {
+	tableList := []string{}
+	for key := range tableNameVsFilePath {
+		tableList = append(tableList, key)
+	}
+
+	target.TableList = strings.Join(tableList, ",")
+}
+
+func parseFileTableMapping() {
+	if fileTableMapping != "" {
+		keyValuePairs := strings.Split(fileTableMapping, ",")
+		for _, keyValuePair := range keyValuePairs {
+			fileName, table := strings.Split(keyValuePair, ":")[0], strings.Split(keyValuePair, ":")[1]
+			tableNameVsFilePath[table] = filepath.Join(dataDir, fileName)
+		}
+	} else {
+		utils.PrintAndLog("Note: --file-table-map flag is not provided, default will assume the file names in format as mentioned in the docs. Refer - link")
+		// get matching file in data-dir
+		files, err := filepath.Glob(filepath.Join(dataDir, "*_data.sql"))
```

Comment:
This code path is specific to `import data file`. The choice we make here is not going to have any impact on `import data` paths.
The most frequent use-case for `import data file` is going to be to import CSV files. So it is better to default for CSV.

Reply:
ok, I believe you just mean to switch to search for `"*_data.sql"` in `data-dir`.

I thought you were saying to change the extension of symlinks also we create. In that case, there will be more changes required because of these assumptions:
```
[root@ip-10-9-80-202 yb_migrate]# grep -i "_data.sql" cmd/importData.go
	so eg. file EMPLOYEE_data.sql -> table EMPLOYEE which needs to converted for using further */
		origDataFile := exportDir + "/data/" + t + "_data.sql"
	dataFilePatern := fmt.Sprintf("%s/*_data.sql", exportDataDir)
	pat := regexp.MustCompile(`.+/(\S+)_data.sql`)
			tables = append(tables, match[1]) //ora2pg data files named like TABLE_data.sql

```

Reply:
done